src/libMessage/Messenger.cpp

```diff
@@ -56,8 +56,204 @@ namespace
         return protoMessage.SerializeToArray(dst.data() + offset,
                                              protoMessage.ByteSize());
     }
+
+    bool SetConsensusAnnouncementCore(
+        ZilliqaMessage::ConsensusAnnouncement& announcement,
+        const uint32_t consensusID, const vector<unsigned char>& blockHash,
+        const uint16_t leaderID, const pair<PrivKey, PubKey>& leaderKey)
+    {
+        LOG_MARKER();
+
+        // Set the consensus parameters
+
+        announcement.mutable_data()->set_consensusid(consensusID);
+        announcement.mutable_data()->set_blockhash(blockHash.data(),
+                                                   blockHash.size());
+        announcement.mutable_data()->set_leaderid(leaderID);
+
+        if (!announcement.data().IsInitialized())
+        {
+            LOG_GENERAL(
+                WARNING,
+                "ConsensusAnnouncement.ConsensusInfo initialization failed.");
+            return false;
+        }
+
+        // Sign the announcement
+
+        vector<unsigned char> tmp;
+
+        if (announcement.has_dsblock()
```

Comment:
could use the protobuf-generated member function `announcement_case()`  to do a `switch` instead of multiple `if-else`

Reply:
Cool, didn't know about this. Will update it. Thanks